### ISSUE TITLE
When a path is in a connect url, should the request body include a trailing slash?

### ISSUE THREAD

**Issue by nuclearace** [daltoniam/Starscream#394]
@daltoniam I'm noticing that socket.io servers aren't picking up WebSockets transports from my socket.io-client. After doing some digging I've found that [this](https://github.com/daltoniam/Starscream/blob/master/Sources/WebSocket.swift#L532) bit of code does not add a trailing slash if the path is not empty. Is this valid, or is this something we should fix here, or do I need to dig around more?

**Comment by nuclearace** [comment]
Example:

```swift
let ws = WebSocket(url: URL(string: "ws://localhost:8080/test/?transport=websocket")!)
```

**Comment by nuclearace** [comment]
It seems whether or not to have a trailing closure depends on the server configuration. However, this code doesn't seem to ever include a trailing / when given a path in the url.

**Comment by daltoniam** [comment]
I personally have seen it both way. I tried to research some of the RFC specs to see what the right behavior was, but best I found was this:

```
 The authority component is preceded by a double slash "//" and is
   terminated by the next slash "/", question-mark "?", or by the end of
   the URI.  Within the authority component, the characters ";", ":",
   "@", "?", and "/" are reserved.
```

https://tools.ietf.org/html/rfc2396#section-3.2

That seems to line up with my personal experience that it can be either way, but that all being said, changing it isn't difficult if we need. I would think we might want to dig deeper though, as it is doesn't seem to fall out side of the RFC and could be something else.

Also I just tested adding `puts "query: #{handshake.query}"` into the `ws-server.rb` in the SimpleTest and it was able to create a proper hash out of them: `query: {"transport"=>"websocket"}`.

**Comment by nuclearace** [comment]
Yeah I agree that this seems like something that should be looked into deeper. Unfortunately I can't update to 3.0 in one of my libs since node's WS server doesn't seem to like differing paths. But adding some config to Starscream to append a trailing slash also feels a bit tacked on.

**Comment by daltoniam** [comment]
I checked in master what should hopefully be a simple fix. I figure it makes more sense to have Starscream make a simple change then try to make every server conform to the expected RFC. Might be a bit tacked on, but from what we found from #392, I'd think it is safer to keep the behavior the same as it use to be versus trying to track down issues with WebSocket servers. Let me know if that fixes it and I will do a 3.0.2 release.
